### PR TITLE
Hackney Address database support

### PR DESCRIPTION
**Pull request by adamtry**
Was used for a script to update assets in the assets DynamoDB table based on addresses in our addresses database (the database consumed by the addresses API)

Includes Makefile to connect to the addresses database and ORM support for HackneyAddress objects.

### CODE REVIEW

File: aws/src/database/domain/dynamo_domain_objects.py [hunk 1]
```diff
@@ -209,28 +209,44 @@ def from_data(cls, data: dict):
...
     assetAddress: AssetAddress | None
     assetCharacteristics: dict | None
-    assetId: str | None
     assetLocation: dict | None
     assetManagement: dict | None
     assetType: str | None
     isActive: int | None
     rootAsset: str | None
     tenure: AssetTenure | None
+    rentGroup: str | None
```
Comment (humulla) [inline aws/src/database/domain/dynamo_domain_objects.py:241]: Why is rent group twice?

File: aws/src/database/rds/addresses/session_for_addresses.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+def session_for_addresses(stage: Stage, expire_on_commit=True, local_port=1433) -> sessionmaker[SA_Session]:
+    """
+    Connect to addresses database on a local port
+    :param stage: Stage to connect to
+    :param expire_on_commit: Select to persist detached objects after transaction commit
+    :return: SQLAlchemy sessionmaker - to be used as a context manager
+    """
+
+    pg_username_path = f"/addresses-api/{stage.to_env_name()}/postgres-username"
```
Comment (humulla) [inline aws/src/database/rds/addresses/session_for_addresses.py:25]: Add a space at the beginning so it works for windows machine. Something like:
`" /addresses-api/{stage.to_env_name()}/postgres-username"`

File: aws/src/database/rds/addresses/session_for_addresses.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    pg_password_path = f"/addresses-api/{stage.to_env_name()}/postgres-password"
```
Comment (humulla) [inline aws/src/database/rds/addresses/session_for_addresses.py:26]: Same as above

File: aws/src/database/rds/addresses/session_for_addresses.py (hunk 2 above)
Comment (adamtry) [reply aws/src/database/rds/addresses/session_for_addresses.py:25]: It should be fine in the Python script as that's not parsed by Powershell / Git Bash

File: aws/src/database/rds/addresses/session_for_addresses.py (hunk 3 above)
Comment (adamtry) [reply aws/src/database/rds/addresses/session_for_addresses.py:26]: As above, should be fine in the Python

File: aws/src/database/domain/dynamo_domain_objects.py (hunk 1 above)
Comment (adamtry) [reply aws/src/database/domain/dynamo_domain_objects.py:241]: Fixed in f7a4811aa95b3836a923dfa17e208f952a6d51b2

**Pull request merged**